source/develop/release-management/features/network/networking-api-security-groups.html.md

```diff
@@ -0,0 +1,352 @@
+* Update a security group
+* Remove a security group
+* Apply a security group to a (set of) port(s)
+* Add a security group rule to a security group
+* Delete security group rule(s) from a security group
+* List security group rules
+* Show a particular security group rule
+* Deny all incoming VM traffic ***if*** configured accordingly
+* Allow all DHCP related traffic ***if*** there's a subnet associated to
+that port's network
+
+### Security groups description
+
+Security groups, in the general sense, are nothing more than containers
+for security group rules. They are applied to logical ports, and thus
+are used to associate a set of rules to a port.
+
+They are are indeed a white list - e.g. everything is *denied* by
+default, and the user adds rules that allow traffic. As long as *one*
+rule is matched, the traffic is allowed.
+
+Security group rules can match on L3 and L4 parameters, specifically on
+the protocol, the IP protocol, and destination ports. The remote IP
+address is also a subject for the match conditions, and rules can be
+added to allow *incoming* and *outgoing* traffic.
+
+More information on the API of security group rules can be found
+[here](https://developer.openstack.org/api-ref/network/v2/#security-group-rules-security-group-rules).
+
+### Activating the feature
+
+Two alternatives are possible:
+
+1. Add a configuration flag to the ovirt-provider-ovn configuration file.
+
+2. Use the [Allowed address pairs](https://developer.openstack.org/api-ref/network/v2/#allowed-address-pairs) networking API extension.
+Using this extension, we would be able to individually specify to which ports we would want to activate the 'deny-all' behavior, instead activating it for all ports.
+
+There's an obvious trade-off between simplicity (and developer hours)
+and granularity of the control.
+
+## Mapping networking API to OVN model objects
+
+The security group data will be model as a [port group](https://github.com/openvswitch/ovs/blob/master/ovn/ovn-nb.xml#L926).
+This ovn-nb table is only available on OVS *master* branch, and will be made available on OVS release 2.10.
+
+This new table maps a list of ACLs to a list of ports, thus eliminating
+the need to replicate the same ACL over and over, having the port in the
+match criteria as the only difference - this way, you match against a
+group of ports. More information can be found [here](http://dani.foroselectronica.es/ovn-profiling-and-optimizing-ports-creation-434/).
+
+### Encoding the security group information
+
+The security group information will be encoded in the port group object.
+
+The security group has the following attributes:
+
+| name                 | type   | description |
+|----------------------|--------|-------------|
+| tenant_id       | string | The ID of the tenant.                                                      |
+| project_id      | string | The ID of the project.                                                     |
```

Comment:
Are `tenant_id` and `project_id` aliases and referring to the same data?

Reply:
In openstack, afaik, a tenant and a project are interchangeable terms. 

Why feature both ?.. Good question; probably it is something they want to change in the future.

Never-the-less, I agree that duplicating the tables from the API doesn't add any value; I did it simply to make life easier for the reader.

If you think it isn't easier, say so and I'll get rid of all tables.

Reply:
tenant_id - user
project_id - one of many projects belonging to one user

Reply:
@mmirecki don't think so; a 'user' is a user; a tenant & project are [interchangeable](https://docs.openstack.org/operations-guide/ops-projects-users.html#projects-or-tenants) terms

Reply:
true, I was not specific enough
tenant_id - organization
user - user
project_id - one of many projects belonging to tenant